### DISCUSSION TITLE
Subscribe/Publish Message bus topic - Need understanding on service communications

### DISCUSSION THREAD

**Discussion by M0hanrajp** [edgexfoundry/community#292] [Q&A] [answered]
### While exploring some logs, I found that in core-data logs there is no log recorded that it is subscribing to an X topic, while mqtt-broker is able to receive the subscribe request from core-data for an x topic. 

I need to know if this is some implementation behavior or I'm missing some conceptual understanding.

**Description**
- Per my understanding, all services will first get configurations from the config provider i.e. core common config bootstrapper.
- Next core-metadata provides information. (May be regarding device profile, device commands any other data).
- Then here X services need to connect to message bus ( Here x in the sense let us consider all services.
- So per the below diagram from documentation [Core-Data, Introduction](https://docs.edgexfoundry.org/3.1/microservices/core/data/Ch-CoreData/)
>Services (like devices services) and other systems can put sensor data on a message bus topic and core data can be configured to subscribed to that topic. This is the default means of getting data to core data. Any service (like an application service or rules engine service) or 3rd system could also subscribe to the same topic. If the sensor data does not need to persisted locally, core data does not have to subscribe to the message bus topic - making core data completely optional.

![image](https://github.com/user-attachments/assets/bc2598b1-3b05-476a-8e02-8d3eb407d25f)

While checking through logs I found that there is no logging recorded for core-data that it is subscribing to edgex/events/device/#, where as I can see that edgex-mqtt-broker has recorded that it is receiving a subscription request to the topic edgex/events/device/#

**So here is some edgex-mqtt broker logs:**
```bash
// Core-data subscribed to mqtt-broker
edgex-mqtt-broker  | 1721115601: New connection from 172.18.0.12:36492 on port 1883.
edgex-mqtt-broker  | 1721115601: New client connected from 172.18.0.12:36492 as core-data (p2, c1, k10).
edgex-mqtt-broker  | 1721115601: No will message specified.
edgex-mqtt-broker  | 1721115601: Sending CONNACK to core-data (0, 0)
edgex-mqtt-broker  | 1721115601: Received SUBSCRIBE from core-data
edgex-mqtt-broker  | 1721115601:        edgex/events/device/# (QoS 0)
edgex-mqtt-broker  | 1721115601: core-data 0 edgex/events/device/#
edgex-mqtt-broker  | 1721115601: Sending SUBACK to core-data
```

Here I want to know if core-data does not record logging for subscribing to events ? because it could see other services are able to subscribe to a particular topic and we can see the logging record for it.

**Comment by M0hanrajp** [comment]
Please let me know if any further information is needed. Thanks

**Comment by cloudxxx8** [comment, marked as answer]
you are right. we are not logging anything about the core-data subscription.
https://github.com/edgexfoundry/edgex-go/blob/9dd757975a2486f93e2289e3629aab843f36bcb7/internal/core/data/controller/messaging/subscriber.go#L32

please feel free to open a PR to add the proper log you want to see.

**Comment by M0hanrajp** [comment]
Hello @cloudxxx8, Thanks for your response.

I'm going through the guidelines for contributing this project from [Commit Message Guidelines](https://github.com/edgexfoundry/edgex-go/blob/c6b1f3a8d30454f4f728b8997d05f0da671aa4be/.github/Contributing.md) & [Committing Code Guidelines](https://wiki.edgexfoundry.org/display/FA/Committing+Code+Guidelines) 
I'm new to GitHub so I'm just reviewing a few things.

Edit: I have created [PR#4840](https://github.com/edgexfoundry/edgex-go/pull/4840)
Thanks.

**Reply by cloudxxx8** [reply]
Thanks, I added my review comment
